openapi/openapi/src/main/scala/endpoints4s/openapi/Endpoints.scala

```diff
@@ -36,143 +37,161 @@ trait EndpointsWithCustomErrors
           )
         }
     val components = Components(
-      schemas = captureSchemas(endpoints),
-      securitySchemes = captureSecuritySchemes(endpoints)
+      schemas = captureSchemas(documentedEndpoints),
+      securitySchemes = captureSecuritySchemes(documentedEndpoints)
     )
     OpenApi(info, items, components)
   }
 
-  type Endpoint[A, B] = DocumentedEndpoint
+  case class Endpoint[A, B](
+      request: Request[A],
+      response: Response[B],
```

Comment:
You need it to be able to implement `mapEndpointResponse` and mapEndpointRequest` required by the new algebra.

Reply:
Yes, but since `type Response[A] = DocumentedResponse`, the operation `mapEndpointResponse` could be implemented as a no-op, no?

Reply:
I didn't think about that, but that makes sense.

Reply:
I can change the `Endpoint` class to be:

```scala
  case class Endpoint[A, B](
      request: DocumentedRequest,
      response: List[DocumentedResponse],
      docs: EndpointDocs,
      securityRequirements: Seq[SecurityRequirement]
  )
```

But I don't think that `mapEndpointResponse` and `mapEndpointRequest` could be made as no-op as the final `DocumentedEndpoint` should reflect the changes made to the request or response.

The thing is that the `Request` and `Response` are "sliced and dices" in the `DocumentedEndpoint` so it would be pretty hard to implement  `mapEndpointResponse` and `mapEndpointRequest` given a `DocumentedEndpoint`. So instead, the `Endpoint` keeps the `Request` and `Response` which makes the implementation of `mapEndpointResponse` and `mapEndpointRequest` trivial and the `DocumentedEndpoint` is a `lazy val` of `Endpoint` so it always reflects the result of these mappings.